### ISSUE TITLE
Gave code Traceback (most recent call last):  File "/app/backup/uiserver.py", line 170, in auth    self._harequests.auth(username, password)  File "/app/backup/harequests.py", line 34, in auth    self._postHassioData("{}auth".format(self.config.get(Setting.HASSIO_URL)), {"username": user, "password": password})  File "/app/backup/harequests.py", line 134, in _postHassioData    return self._validateHassioReply(self._client.post(url, headers=self._getHassioHeaders(), json=json, data=data))  File "/app/backup/harequests.py", line 96, in _validateHassioReply    resp.raise_for_status()  File "/usr/lib/python3.6/site-packages/requests/models.py", line 940, in raise_for_status    raise HTTPError(http_error_msg, response=self)requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://hassio/auth

### ISSUE THREAD

**Issue by craigd86** [sabeechen/hassio-google-drive-backup#63]
Running Hassio in Virtualbox version 0.93.2

Successfully installed Google Drive Backup and had no errors with backups. The system crashed so a new version of Hassio was installed in a new vm. Installed Google Drive Backup in the new, clean version of Hassio to restore from a snapshot. Successfully restored from the snapshot but was given this error upon logging into the UI of Google Drive Backup. Utilized "File This Issue on Github" link provided in the UI.

No problematic logs form Hassio Supervisor or any other location.

Traceback (most recent call last):
  File "/app/backup/coordinator.py", line 102, in _sync
    self._buildModel().sync(self._time.now())
  File "/app/backup/model.py", line 116, in sync
    raise SimulatedError(self.simulate_error)
backup.exceptions.SimulatedError: 
Traceback (most recent call last):
  File "/app/backup/uiserver.py", line 170, in auth
    self._harequests.auth(username, password)
  File "/app/backup/harequests.py", line 34, in auth
    self._postHassioData("{}auth".format(self.config.get(Setting.HASSIO_URL)), {"username": user, "password": password})
  File "/app/backup/harequests.py", line 134, in _postHassioData
    return self._validateHassioReply(self._client.post(url, headers=self._getHassioHeaders(), json=json, data=data))
  File "/app/backup/harequests.py", line 96, in _validateHassioReply
    resp.raise_for_status()
  File "/usr/lib/python3.6/site-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://hassio/auth

Gave code 
Traceback (most recent call last):
  File "/app/backup/uiserver.py", line 170, in auth
    self._harequests.auth(username, password)
  File "/app/backup/harequests.py", line 34, in auth
    self._postHassioData("{}auth".format(self.config.get(Setting.HASSIO_URL)), {"username": user, "password": password})
  File "/app/backup/harequests.py", line 134, in _postHassioData
    return self._validateHassioReply(self._client.post(url, headers=self._getHassioHeaders(), json=json, data=data))
  File "/app/backup/harequests.py", line 96, in _validateHassioReply
    resp.raise_for_status()
  File "/usr/lib/python3.6/site-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://hassio/auth

**Comment by craigd86** [comment]
This was solved. I stopped the add-on and then restarted it back up. No longer receiving any errors.

**Comment by sabeechen** [comment]
Your install has gotten into what I can only describe as a "very weird" state.  I suspect that just restarting the add-on will resolve it, but I can't imagine what would cause it.

Did you do a "Full" restore or a partial one?  Restoring can (ironically) restore an older version of the add-on, so make sure you're still running the latest version of the add-on.

**Comment by craigd86** [comment]
I did do a full restore. I read the FAQs after I submitted the issue and of course, finding the answer by restarting the add-on did the trick.

**Comment by Sjeff** [comment]
I receive the same error.
Didn't do any restore.
After a addon restart the error is gone but will come back the next day.

```
Gave code 
Traceback (most recent call last):
  File "/app/backup/uiserver.py", line 170, in auth
    self._harequests.auth(username, password)
  File "/app/backup/harequests.py", line 34, in auth
    self._postHassioData("{}auth".format(self.config.get(Setting.HASSIO_URL)), {"username": user, "password": password})
  File "/app/backup/harequests.py", line 134, in _postHassioData
    return self._validateHassioReply(self._client.post(url, headers=self._getHassioHeaders(), json=json, data=data))
  File "/app/backup/harequests.py", line 96, in _validateHassioReply
    resp.raise_for_status()
  File "/usr/lib/python3.6/site-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://hassio/auth
```
Logfile: [hassio-google-drive-backup.log](https://github.com/sabeechen/hassio-google-drive-backup/files/3230976/hassio-google-drive-backup.log)

**Comment by sabeechen** [comment]
Do you have require_login set to true in your config?  I'm struggling to figure out how this happens.

**Comment by Sjeff** [comment]
Current addon config:
```
{
  "max_snapshots_in_hassio": 10,
  "max_snapshots_in_google_drive": 24,
  "days_between_snapshots": 1,
  "use_ssl": false,
  "verbose": true,
  "snapshot_stale_minutes": 250,
  "require_login": true,
  "snapshot_time_of_day": "02:15",
  "generational_days": 7,
  "generational_weeks": 4,
  "generational_day_of_week": "fri",
  "generational_months": 12,
  "generational_day_of_month": 1,
  "send_error_reports": false
}
```

**Comment by sabeechen** [comment]
I figured it out, "failed" logins to the UI hit a bad code path that would cause sync to fail until a restart, even if a subsequent login succeeded.  I've fixed it in v0.97.4, which I just released.  

Click the refresh icon at the top right of the add-on store page if you don't see the new version yet.
